config/v1/types_infrastructure.go

```diff
@@ -500,6 +505,58 @@ type AWSResourceTag struct {
 	Value string `json:"value"`
 }
 
+// DNSAndLBConfiguration contains information about API and API-Int Load Balancers
+// and who is responsible for that configuration.
+// +union
+type DNSAndLBConfiguration struct {
+	// provider determines which DNS solution is in use for this cluster.
+	// When the user has pre-created the Load Balancers for API and API-Int and the
+	// the DNS solution has already been configured, this field is set to "UserProvided".
+	// When the cluster is resposible for its own DNS configuration and LB creation,
+	// this field is set to "ClusterProvided".
+	// +default="ClusterProvided"
+	// +kubebuilder:default:="ClusterProvided"
+	// +kubebuilder:validation:Enum="UserProvided";"ClusterProvided"
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	Provider DNSAndLBProviderType `json:"provider,omitempty"`
+
+	// userProvided contains information about the user created LoadBalancers.
+	// These values are expected to be set only when provider is set to "UserProvided"
+	UserProvided *LBConfigType `json:"userProvided,omitempty"`
+
+	// clusterProvided contains information about the OpenShift cluster created LoadBalancers.
+	ClusterProvided *LBConfigType `json:"clusterProvided,omitempty"`
```

Comment:
If you don't want to have config for ClusterProvided we don't need to have an option here